### PR TITLE
feat: --proxy: forward /_/ to a specified url

### PR DESCRIPTION
**Pull request by ericswanson-dfinity**
Add command-line parameter `--proxy <url>` which, if present, forwards requests with path beginning with `/_/` to the specified endpoint.

Used by `dfx` to serve data from  `/_/candid`

### CODE REVIEW

File: icx-proxy/src/main.rs [hunk 1]
```diff
@@ -418,17 +422,26 @@ async fn handle_request(
...
-    match if request.uri().path().starts_with("/api/") {
+    let request_uri_path = request.uri().path();
+    match if request_uri_path.starts_with("/api/") {
         slog::debug!(
             logger,
             "URI Request to path '{}' being forwarded to Replica",
             &request.uri().path()
         );
         forward_api(&ip_addr, request, &replica_url).await
+    } else if proxy_url.is_some() && request_uri_path.starts_with("/_/") {
```
Comment (hansl) [inline icx-proxy/src/main.rs:438]: Need to return 404 if there's no `proxy_url` configured.

Comment (hansl) [reply icx-proxy/src/main.rs:438]: Ah I see we didn't do that before. I won't block this PR on it then.

Comment (ericswanson-dfinity) [reply icx-proxy/src/main.rs:438]: I'm also not clear on how this is supposed to work outside of `dfx`

Comment (ericswanson-dfinity) [reply icx-proxy/src/main.rs]: Anyway, this is updated to return 404 now

**Review by hansl — APPROVED** [review]
LGTM. Loves adding 32 lines to make this happen! This is awesome

**Pull request merged**